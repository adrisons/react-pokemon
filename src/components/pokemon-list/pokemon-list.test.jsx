import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import PokemonList from "./pokemon-list";
describe("GIVEN: PokemonList", () => {
  describe("WHEN: PokemonList with no pokemons", () => {
    const component = renderer.create(<PokemonList />);
    const tree = component.toJSON();
    it("THEN: should match snapshot", () => {
      expect(tree).toMatchSnapshot();
    });

    it("THEN: component should display no pokemons found message", () => {
      expect(tree.children.length).toEqual(1);
      expect(tree.children[0].props.className).toEqual("not-found");
    });
  });

  describe("WHEN: PokemonList with pokemons", () => {
    const pokemonsMock = [
      {
        name: "bulbasaur",
        url: "https://pokeapi.co/api/v2/pokemon/1/",
      },
      {
        name: "ivysaur",
        url: "https://pokeapi.co/api/v2/pokemon/2/",
      },
      {
        name: "venusaur",
        url: "https://pokeapi.co/api/v2/pokemon/3/",
      },
    ];
    const component = renderer.create(
      <Router>
        <PokemonList pokemons={pokemonsMock} />
      </Router>
    );
    const tree = component.toJSON();
    it("THEN: should match snapshot", () => {
      expect(tree).toMatchSnapshot();
    });

    it("THEN: component should display pokemons list", () => {
      expect(tree.children.length).toEqual(3);
    });
    it("THEN: each element should have a link with its ID", () => {
      expect(tree.children[0].children[0].props.href).toEqual("/detail/1");
      expect(tree.children[1].children[0].props.href).toEqual("/detail/2");
      expect(tree.children[2].children[0].props.href).toEqual("/detail/3");
    });
    it("THEN: each element should display pokemon name and ID", () => {
      const linkObj = tree.children[0].children[0];
      expect(linkObj.children[0].children.join("")).toEqual("#1");
      expect(linkObj.children[1].children.join("")).toEqual("bulbasaur");
    });
  });
});
