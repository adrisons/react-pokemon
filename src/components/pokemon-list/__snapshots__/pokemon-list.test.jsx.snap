// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN: PokemonList WHEN: PokemonList with no pokemons THEN: should match snapshot 1`] = `
<ul
  className="list-group"
>
  <div
    className="not-found"
  >
    No pokemons found
  </div>
</ul>
`;

exports[`GIVEN: PokemonList WHEN: PokemonList with pokemons THEN: should match snapshot 1`] = `
<ul
  className="list-group"
>
  <li>
    <a
      className="option"
      href="/detail/1"
      onClick={[Function]}
    >
      <div
        className="id"
      >
        #
        1
      </div>
      <div
        className="name"
      >
        bulbasaur
      </div>
    </a>
  </li>
  <li>
    <a
      className="option"
      href="/detail/2"
      onClick={[Function]}
    >
      <div
        className="id"
      >
        #
        2
      </div>
      <div
        className="name"
      >
        ivysaur
      </div>
    </a>
  </li>
  <li>
    <a
      className="option"
      href="/detail/3"
      onClick={[Function]}
    >
      <div
        className="id"
      >
        #
        3
      </div>
      <div
        className="name"
      >
        venusaur
      </div>
    </a>
  </li>
</ul>
`;
